commit 2f2ead61c15e59a352b23f135848f0d0bb0398fa
Author: Wu Zheng <wu.zheng@intel.com>
Date:   Sun Sep 29 16:32:12 2013 +0800

    Stop and clear the pending transfers in a session if a transfer fails

diff --git a/obexd/client/session.c b/obexd/client/session.c
index 331d23d..24e652d 100644
--- a/obexd/client/session.c
+++ b/obexd/client/session.c
@@ -831,8 +831,17 @@ static void session_terminate_transfer(struct obc_session *session,
 
 	pending_request_free(p);
 
+#ifdef __TIZEN_PATCH__
+	if (session->p == NULL) {
+		if (gerr)
+			obc_session_shutdown(session);
+		else
+			session_process_queue(session);
+	}
+#else
 	if (session->p == NULL)
 		session_process_queue(session);
+#endif
 
 	obc_session_unref(session);
 }
